c_api.c

```diff
@@ -210,9 +208,14 @@ session_key_t *get_session_key_by_ID(unsigned char *target_session_key_id,
                 "Failed to send_session_key_request_check_protocol().");
             return NULL;
         }
-        s_key = s_key_list->s_key;
-        add_session_key_to_list(s_key, existing_s_key_list);
-        free(s_key_list);
+        int index =
+            add_session_key_to_list(s_key_list->s_key, existing_s_key_list);
+        if (index < 0) {
+            SST_print_error("Failed to add_session_key_to_list().");
+            return NULL;
+        }
+        free_session_key_list_t(s_key_list);
+        s_key = &existing_s_key_list->s_key[index];
```

Comment:
This part is changes for the memory issues. The `s_key` is now pointed from the `existing_s_key_list`.